packages/graph/src/-private/-diff.ts

```diff
@@ -70,11 +75,43 @@ function _deprecatedCompare<T>(
       // detect reordering, adjusting index for duplicates
       // j is always less than i and so if i < prevLength, j < prevLength
       if (member !== prevState[j]) {
-        changed = true;
-      } else if (!changed && j < priorLocalLength) {
-        const priorLocalMember = priorLocalState![j];
-        if (priorLocalMember !== member) {
-          changed = true;
+        // the new remote order does not match the current remote order
+        // indicating a change in membership or reordering
+        remoteOrderChanged = true;
+        // however: if the new remote order matches the current local order
+        // we can disregard the change notification generation so long as
+        // we are not configured to reset on remote update (which is deprecated)
+        if (DEPRECATE_RELATIONSHIP_REMOTE_UPDATE_CLEARING_LOCAL_STATE) {
+          if (!remoteClearsLocal && i < priorLocalLength) {
+            const priorLocalMember = priorLocalState![j];
+            if (priorLocalMember !== member) {
+              changed = true;
+            }
+          } else {
+            changed = true;
+          }
+        } else {
+          if (i < priorLocalLength) {
+            const priorLocalMember = priorLocalState![j];
+            if (priorLocalMember !== member) {
+              changed = true;
+            }
+          } else {
+            changed = true;
+          }
```

Comment:
@runspired Why do we need this if else when you are performing same operations inside ? 
Can we just do it with just one if condition like below 
```
if (DEPRECATE_RELATIONSHIP_REMOTE_UPDATE_CLEARING_LOCAL_STATE) {
if ((!remoteClearsLocal && i < priorLocalLength) || i < priorLocalLength || remoteClearsLocal ) {
           if (!changed && j < priorLocalLength) {
             const priorLocalMember = priorLocalState![j];
              changed  = priorLocalMember !== member ? true : false;
           }
         }
       }
}

```

Reply:
`!remoteClearsLocal && ` is the difference. The way we approach deprecation flagging (so that deprecated code can be fully eliminated) is to wrap the full version of the deprecated code in its flag and the else be the full new version.

RE `changed  = priorLocalMember !== member ? true : false;`, we don't do this because once `changed` is true we never revert it to false.

Reply:
@runspired ok , in that case make it as `changed  = priorLocalMember !== member && true ;`